sdk/core/azure-core/azure/core/pipeline/policies/_base.py

```diff
@@ -50,6 +50,11 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+def safe_deep_copy(x):
+    try:
+        return copy.deepcopy(x)
+    except ValueError:
+        return copy.copy(x)
```

Comment:
Just to clarify this is not what I was suggesting. :)  I was proposing to make it so that 
```
copy.deepcopy(http_request)
```
always works.  And the mechanism I think would allow this is to implement a custom `__deepcopy__` special method on the request class that knows how to ignore/discard un-copyable attributes like streams when performing deep copies.